src/test/testDirectives.ts

```diff
@@ -1039,10 +1039,9 @@ describe('@directives', () => {
       directives: {
         objectTypeDirective: class extends SchemaDirectiveVisitor {
           public visitObject(object: GraphQLObjectType) {
-            HumanType = Object.create(object, {
+            return HumanType = Object.create(object, {
               name: { value: 'Human' }
             });
-            this.schema.getTypeMap()[object.name] = HumanType;
```

Comment:
This snippet is a good demonstration of the value of this functionality. Mutating the type map doesn't work here, because it has no effect on the rest of the traversal, whereas returning a replacement object immediately determines what gets visited next.